### PR TITLE
Add delete button for additionalProperties key-value pair

### LINKED ISSUE #1099: multiple bugs with additionalProperties component
### Prerequisites

- [x] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.

### Description

- There is no option to remove key-value pair once it's added
- Key changes name when doubleclicked
- Key-value pair also sometimes change position, when you add more pairs, change their keys and doubleclick again

### Steps to Reproduce

1. Loot at provided jsfiddle
2. Doubleclick "newKey Key"
3. It changes name to "newKey-1 Key"
4. Doubleclick again
5. "newKey-1-1 Key"
6. add one more pair
7. click key again
8. they should change their order

https://jsfiddle.net/aerfio/mynuxdc4/

#### Expected behavior

I expect to have option to remove once added key-value pair and to preserve order of those pairs.

#### Actual behavior

No "remove/delete/X" button, ordering is not preserved 

### Version

1.0.6

Related: #228

### PR DESCRIPTION
**Pull request by aerfio**
### Reasons for making this change

This PR adds delete button for key-value pair in _additionalProperties_, also rearanges those two inputs + button to be in line, not one below another. It also removes the ability to remove pair by removing content of value input - we have button for that. 

I have also fixed tests related to time widget (change from 2018 to 2019 made two tests crash).

Work largely based on this PR: #1085, but since nothing has changed there for some time I decided to make this PR. 

This PR would resolve #1099.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [x] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by aerfio** [comment]
@edi9999 @glasserc Can you have a look at this?

**Comment by aerfio** [comment]
@glasserc Existence of `addedByAdditionalProperties` flag feels a bit hacky, but it's used _only_ in additionalProperties, and without it emptying a value input makes pair disappear, which is counter-intuitive -> we have a delete button introduced in this PR for this.

If emptying a string would not make pair disappear it would be great, but I couldn't find a better place for it - if you know a better place/method how to make it work like it should then please let me know 👍 . 

After this PR gets merged when can we expect next release?

**Comment by glasserc** [comment]
Thanks for your contribution. I agree that this is maybe the least bad of the possible solutions to clearing out an input field. I'll cut a release later today.

**Comment by pieplu** [comment]
Hi! @aerfio , as I say on this comment: https://github.com/mozilla-services/react-jsonschema-form/pull/1123/files#r249176646

@glasserc This PR merged include on v1.2.0 useless div, and break some styles

File: src/components/fields/SchemaField.js [hunk 1]
```diff
@@ -121,35 +121,57 @@ function DefaultTemplate(props) {
...
+            />
+          )}
         </div>
-      )}
-      {displayLabel && <Label label={label} required={required} id={id} />}
-      {displayLabel && description ? description : null}
-      {children}
-      {errors}
-      {help}
+      </div>
```
[the remark this answers is not in the document]
Comment (aerfio) [reply src/components/fields/SchemaField.js]: @pieplu I agree that lines 134 and 135 of this file can surely be merged into one (mistake on my side), which would probably leave just one empty div which is in line 150, but I at the first glance I don't know how to achieve same appearence using Bootstrap 3, and have a look at this: https://github.com/mozilla-services/react-jsonschema-form/pull/1123#discussion_r246505297

[the remark this answers is not in the document]
Comment (pieplu) [reply src/components/fields/SchemaField.js]: Ok, I can try PR a fix with bootstrap style

**Pull request merged**